packages/website/src/components/linter/CompilerHost.ts

```diff
@@ -0,0 +1,61 @@
+import type { ScriptKind, System, SourceFile, CompilerHost } from 'typescript';
+
+function getScriptKind(
+  ts: typeof import('typescript'),
+  filePath: string,
+): ScriptKind {
+  const extension = (/(\.[a-z]+)$/.exec(filePath)?.[0] ?? '').toLowerCase();
+
+  switch (extension) {
+    case '.ts':
+      return ts.ScriptKind.TS;
+    case '.tsx':
+      return ts.ScriptKind.TSX;
+
+    case '.js':
+      return ts.ScriptKind.JS;
+    case '.jsx':
+      return ts.ScriptKind.JSX;
+
+    case '.json':
+      return ts.ScriptKind.JSON;
+
+    default:
+      // unknown extension, force typescript to ignore the file extension, and respect the user's setting
+      return ts.ScriptKind.TS;
+  }
+}
```

Comment:
I was actually going to expose this from `typescript-estree` in #5027
Is there a reason that you pass in `ts` specifically instead of using `window.ts`?

Reply:
no, there is no difference, both properties have same value